kolibri/deployment/default/custom_django_cache.py

```diff
@@ -0,0 +1,141 @@
+import sqlite3
+
+from diskcache import DjangoCache
+from django.core.cache.backends.base import DEFAULT_TIMEOUT
+
+
+class CustomDjangoCache(DjangoCache):
+    """
+    Inherits from the DjangoCache to better manage the error handling of
+    the diskcache package by try-catching methods that perform database operations
+    under the hood by compairing against the version of diskcache used in kolibri:
+
+    https://github.com/grantjenks/python-diskcache/blob/v4.1.0/diskcache/djangocache.py
+    """
+
+    def add(
+        self,
+        key,
+        value,
+        timeout=DEFAULT_TIMEOUT,
+        version=None,
+        read=False,
+        tag=None,
+        retry=True,
+    ):
+        try:
+            return DjangoCache.add(self, key, value, timeout, version, read, tag, retry)
```

Comment:
It doesn't make a huge difference, but I think in general I'd prefer we use `super` for these calls, so this would become:

```
return super(CustomDjangoCache, self).add(key, value, timeout, version, read, tag, retry)
```

Main thing to note is that we need to pass the class and self arguments to `super` for Python 2.7 compatibility, and that we then don't have to pass `self` to the method invocation, as it's already bound.

It's also possible that we could make this DRYer by using a decorator that wraps each of the methods, but this explicitness is also nice.

Reply:
This change has been effected.